### PR TITLE
Added string-concat and string-join primitives

### PR DESCRIPTION
**Pull request by kjmcqueen**
https://rchain.atlassian.net/browse/ROS-253
https://rchain.atlassian.net/browse/ROS-306

### CODE REVIEW

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala [hunk 1]
```diff
@@ -44,6 +44,69 @@ object rblstring {
...
+    *  (string-join 0 ":" ["123" "456" "789"]) ===> "123:456:789"
+    *  (string-join 1 ":" ["123" "456" "789"]) ===> ":123:456:789"
+    *  (string-join 2 ":" ["123" "456" "789"]) ===> "123:456:789:"
+    *  (string-join 3 ":" ["123" "456" "789"]) ===> ":123:456:789:"
+    *  (string-join 3 ":" []) ===> ""
+    */
+  object stringJoin extends Prim {
+    override val name: String = "string-join"
+    override val minArgs: Int = 3
+    override val maxArgs: Int = 3
```
Comment (tymm) [inline roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: Missing newline.

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala [hunk 2]
```diff
@@ -44,6 +44,69 @@ object rblstring {
...
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem = ctxt.argvec.elem
+
+      checkFixnum(0, elem).flatMap( // Ensure arg0 is a Fixnum
```
Comment (tymm) [inline roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: Can you please use a for-expression here?
```
for {
  code <- checkFixnum(0, elem)
  sep <- checkRblString(1, elem)
  ...
} yield {
  val strings = ...
}
````

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala (hunk 1 above)
Comment (kjmcqueen) [reply roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: Done

File: roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala (hunk 2 above)
Comment (kjmcqueen) [reply roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala]: I like that.  I wish you'd taught me that earlier.  ;-)

**Pull request merged**